### PR TITLE
fix missing supported res type in webapp

### PR DESCRIPTION
**Pull request by zhiyuli**
This pr was to fix issue #1583 .

We removed the hard-coded "supported resource type" options. The revised webapp res type is able to dynamically harvest all existing resource types from system and use them to build "supported resource type" options.

### CODE REVIEW

File: hs_tools_resource/page_processors.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def landing_page(request, page):
                 supported_res_types.first().get_supported_res_types_str()
             supported_res_types_array = supported_res_types_str.split(',')
             for type_name in supported_res_types_array:
-                for display_name_tuple in SupportedResTypes_choices:
+                for display_name_tuple in get_SupportedResTypes_choices():
```
Comment (pkdash) [inline hs_tools_resource/page_processors.py]: The variable name ```display_name_tuple``` seems confusing as the iteration is over a list of list items.

File: hs_tools_resource/utils.py [hunk 2]
```diff
@@ -27,3 +28,25 @@ def parse_app_url_template(url_template_string, term_dict_list=()):
...
+        ["RESOURCE_CLASS_NAME_N", "RESOURCE_VERBOSE_NAME_N"],
+    ]
+    """
+
+    result_list = []
+    res_types_list = get_resource_types()
+    for r_type in res_types_list:
+        class_name = r_type.__name__
+        verbose_name = r_type._meta.verbose_name
+        if "toolresource" not in class_name.lower():
```
Comment (pkdash) [inline hs_tools_resource/utils.py]: Should it be ```if class_name.lower()  != "toolresource":```

**Comment by zhiyuli** [comment]
@pkdash 
comments have been addressed. thanks

**Comment by pkdash** [comment]
+1

**Pull request merged**